app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -1191,6 +1234,209 @@ class BrowserTabFragment : Fragment(), FindListener, CoroutineScope {
         }
     }
 
+    inner class BrowserTabFragmentExperimentDecorator {
```

Comment:
I like how we moved all the UI logic related to our bottom bar experiment inside this `Decorator`, but I see outside this class some logic related to our experiment, so it's not clear to me when we delegate to this decorator and when not.

For example, how to increment the tab count is outside the decorator whereas visibility or clickListener are inside this class.

Reply:
that's a good point, I've moved the tabsCount animation to the ``Decorator` for simplicity